test/shell-pipeline-tests/common.sh

```diff
@@ -75,10 +75,93 @@ function createImageRegistrySecret() {
+  NEW_DIGEST=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.status.results[?\(@.name==\'image-sha\'\)].value})
+  BASE_REF_NAME=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitRepoBranchBase\'\)].value})
+  GIT_SERVER=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitServer\'\)].value})
+  GIT_ORG_NAME=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitOrgName\'\)].value})
+  GIT_REPO_NAME=$(oc get pipelinerun "$PIPELINE_RUN_NAME" -o jsonpath={.spec.params[?\(@.name==\'gitRepoName\'\)].value})
+  GIT_URL="$GIT_SERVER/$GIT_ORG_NAME/$GIT_REPO_NAME"
+
+  # Actual values
+  PR_PIPELINE_RUN_NAME=$(grep "PipelineRun Name" "$PR_INFO_PATH" )
+  PR_PIPELINE_RUN_UID=$(grep "PipelinRun UID" "$PR_INFO_PATH")
+  PR_FILE_PATH=$(grep "File Path" "$PR_INFO_PATH") # TODO Remove
+  PR_IMAGE_REGISTRY=$(grep "Image registry" "$PR_INFO_PATH")
+  PR_NEW_DIGEST=$(grep "New Digest" "$PR_INFO_PATH")
+  PR_BASE_REF_NAME=$(gh pr view "$PR_URL" --json baseRefName -q ".baseRefName")
+  PR_HEAD_REF_NAME=$(gh pr view "$PR_URL" --json headRefName -q ".headRefName")
+  PR_DIFF_DIGEST=$(grep "+  digest" "$PR_DIFF_PATH")
+  PR_DIFF_NEW_NAME=$(grep "newName" "$PR_DIFF_PATH")
+
+  # Assertions
+  echo "Running Pull Request Assertions"
+  assertExitCode=0
+  assertContains "$PR_PIPELINE_RUN_NAME" "$PIPELINE_RUN_NAME"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_PIPELINE_RUN_UID" "$PIPELINE_RUN_UID"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_FILE_PATH" "$FILE_PATH"; (( assertExitCode = assertExitCode || $? )) # TODO Remove
+  assertContains "$PR_IMAGE_REGISTRY" "$IMAGE_REGISTRY"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_NEW_DIGEST" "$NEW_DIGEST"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_BASE_REF_NAME" "$BASE_REF_NAME"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_URL" "$GIT_URL"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_HEAD_REF_NAME" "pipeline_$PIPELINE_RUN_UID"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_DIFF_DIGEST" "$NEW_DIGEST"; (( assertExitCode = assertExitCode || $? ))
+  assertContains "$PR_DIFF_NEW_NAME" "$IMAGE_REGISTRY"; (( assertExitCode = assertExitCode || $? ))
+
+  return "$assertExitCode"
+}
+
+function getGitHubToken() {
+  AI_EDGE_GITHUB_VAULT_OPENSHIFT_CI_SECRET_PATH="${CUSTOM_GIT_CREDENTIALS_SECRET_PATH:-/secrets/ai-edge-github}"
+  GH_TOKEN=$(cat "$AI_EDGE_GITHUB_VAULT_OPENSHIFT_CI_SECRET_PATH"/token)
+  echo "$GH_TOKEN"
+}
+
+function installGithubCLI() {
+  echo "Installing GitHub CLI"
+  wget -nv https://github.com/cli/cli/releases/download/v2.47.0/gh_2.47.0_linux_amd64.tar.gz
+  tar -xf gh_2.47.0_linux_amd64.tar.gz
+  export PATH="$(pwd)/gh_2.47.0_linux_amd64/bin:$PATH"
+  echo "Using GitHub CLI from $(which gh)"
+}
```

Comment:
Notice the marker that GitHub (and even Git) put here at the end of the file when it's missing a newline at the end.

It's good practice to always have a newline at the end of files if possible, and you can typically configure your editor/IDE to ensure it's there. See this SO answer for more details: https://stackoverflow.com/a/729795